### ISSUE TITLE
Curation Settings missing in Elastic Deployment Resource

### ISSUE THREAD

**Issue by ggsood** [elastic/terraform-provider-ec#166]
Curation settings is missing in the Elastic Deployment Resource.

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed
- [x] I am reporting the issue to the correct repository (for multi-repository projects)

## Expected Behavior
Optional Curation settings should be present when configuring an ES Deployment.
This is important when template type is hot-warm or hot-warm-v2

## Current Behavior
Curation settings are currently missing in the ES resource, absence of this setting causes the following issues
1. Converts Hot Warm deployments to curation based setting - which will soon be deprecated
2. Cannot enable migration from the Cloud UI, the option does not work 
3. Disables ILM option in the Kibana
4. Does not adhere to the API JSON generated from UI, example

```json
POST /api/v1/deployments?validate_only=false
{
  "resources": {
    "elasticsearch": [
      {
        "region": "aws-eu-west-2",
        **"settings": {
          "curation": {
            "specs": []
          },**
          "dedicated_masters_threshold": 6
        },
        "plan": {
          "cluster_topology": [
            {
              "elasticsearch": {
                "node_attributes": {
                  "data": "hot"
                },
                "enabled_built_in_plugins": []
              },
              "node_type": {
                "data": true,
                "master": true,
                "ingest": true
              },
              "instance_configuration_id": "aws.data.highio.i3",
              "zone_count": 1,
              "size": {
                "value": 1024,
                "resource": "memory"
              }
            },
            {
              "elasticsearch": {
                "node_attributes": {
                  "data": "warm"
                },
                "enabled_built_in_plugins": []
              },
              "node_type": {
                "data": true,
                "master": false,
                "ingest": true
              },
              "instance_configuration_id": "aws.data.highstorage.d2",
              "zone_count": 1,
              "size": {
                "value": 2048,
                "resource": "memory"
              }
            }
          ],
          "elasticsearch": {
            "version": "7.9.3"
          },
          "deployment_template": {
            "id": "aws-hot-warm-v2"
          }
        },
        "ref_id": "main-elasticsearch"
      }
    ],
    "enterprise_search": [],
    "kibana": [
      {
        "elasticsearch_cluster_ref_id": "main-elasticsearch",
        "region": "aws-eu-west-2",
        "plan": {
          "cluster_topology": [
            {
              "instance_configuration_id": "aws.kibana.r5d",
              "zone_count": 1,
              "size": {
                "resource": "memory",
                "value": 1024
              }
            }
          ],
          "kibana": {
            "version": "7.9.3"
          }
        },
        "ref_id": "main-kibana"
      }
    ],
    "apm": []
  },
  "name": "hot-warm-architecture-deployment",
  "metadata": {
    "system_owned": false
  }
}
```

## Possible Solution
Add the settings optional block in the terraform code with option to configure curation settings
- Curation setting to left blank, this will handle post depreciation
- Curation setting with spec info , for deployments using curation

## Steps to Reproduce
2 ways of reproducing

1. Create a hot warm architecture using the terraform resource
2. Curation is enabled by default
3. Try to manually change it to ILM , UI gets struck at this point
4. Go to Kibana ILM option is not available

-
1. For an existing Hot warm deployment having ILM enabled
2. Import the cluster in terraform using template hot-warm
3. Perform any update using terraform, curation is enabled and ILM is disabled.

## Context
Enables curation setting for a Hot Warm Deployment, cannot enable ILM for the cluster after that.

## Your Environment
Elastic Cloud

**Comment by ggsood** [comment]
Hi All 

On further investigations i found that the curation settings are passed from the templates for hot-warm

https://api.elastic-cloud.com/api/v1/deployments/templates

```json
                          "elasticsearch": {
                                "curation": {
                                    "from_instance_configuration_id": "aws.data.highio.i3",
                                    "to_instance_configuration_id": "aws.data.highstorage.d2"
                                }
                            }
                        },
                        "settings": {
                            "curation": {
                                "specs": [
                                    {
                                        "index_pattern": "logstash-*",
                                        "trigger_interval_seconds": 86400
                                    },
                                    {
                                        "index_pattern": "filebeat-*",
                                        "trigger_interval_seconds": 86400
                                    },
                                    {
                                        "index_pattern": "metricbeat-*",
                                        "trigger_interval_seconds": 86400
                                    }
                                ]
                            },
                            "dedicated_masters_threshold": 6
                        }
                    }
```

**Comment by ggsood** [comment]
Is fixed by PR #167
